### PR TITLE
Feature to capture additional fields in credit form

### LINKED ISSUE #329: Add new "Credit card" account form
We're working on adding each new account type as they've got slightly different "field" needs.

All accounts have "Name" and "Balance" by default.

Additional fields:
* Type — Select (Mastercard, Visa, American Express, Discover)
* Limit — Number

View is in `/views/accounts/account/_credit.html.erb`

### PR DESCRIPTION
**Pull request by sarvaiyanidhi**
Closes #329

This PR implements below changes

- Migration to add news fields in account_credits table
- Enabling nested attributes on a delegated_type
- Capturing nested_attribute in create action of Account

### CODE REVIEW

**Comment by sarvaiyanidhi** [comment]
@Shpigford I have question here related to subtype field defined in app/views/accounts/account/_depository.html.erb
Is that field specific to account_depositories table? If so, why is it added in accounts table and in case it is for accounts table so can it be part of new.html.erb

Not sure the solution I followed of nested_attribute for delegate_type is correct approach but due to this solution subtype field is not saving up in database as it is considering it as nested attribute.

Please advice on implementation if I need to change the logic.

**Comment by sarvaiyanidhi** [comment]
Just referred implementation done for #343 .. If that is the case I think I complicated requirement assuming we need to have separate fields to that type

**Comment by jespr** [comment]
I think `subtype` is used as the "subtype" for the account. So for accounts where type is "credit", subtype can be any of: Mastercard, Visa, American Express, Discover

**Comment by sarvaiyanidhi** [comment]
Yeah, got it I thought it needs to be separate fields related to specific table and complicated used nested_attribute form concept

**Comment by jespr** [comment]
I like the way you've abstracted the types to the model - I think I'm going to do that in my PR as well!

**Comment by sarvaiyanidhi** [comment]
I think let us wait for input from @Shpigford as not sure if what I implemented is required as depository type also follows your approach,

**Comment by Shpigford** [comment]
subtypes just help specify the type.

So "depository" is just the high level with subtypes of "checking" and "savings" for instance.

And "investment" would have subtypes like "401k" and "roth ira"

File: app/views/accounts/account/_credit.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div class="relative p-4 border border-gray-100 bg-offwhite rounded-xl focus-within:bg-white focus-within:shadow focus-within:opacity-100">
+    <label for="type" class="block text-sm font-medium opacity-50 focus-within:opacity-100">Type</label>
```
Comment (ricsdeol) [inline app/views/accounts/account/_credit.html.erb:2]: Should use rails form help `ff.label :card_type`

And use internatiolizations...

**Comment by sarvaiyanidhi** [comment]
Thank you for clarification I will just push new PR as this one is complicated by adding new attributes to account_credits table

File: app/views/accounts/account/_credit.html.erb (hunk 1 above)
Comment (sarvaiyanidhi) [reply app/views/accounts/account/_credit.html.erb:2]: Thanks @ricsdeol Noted. Will do in new PR

File: app/models/account/credit.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
 class Account::Credit < ApplicationRecord
   include Accountable
+
+  CARD_TYPES = %i[mastercard visa american_express discover].freeze
```
Comment (ricsdeol) [inline app/models/account/credit.rb:4]: Coukd use rails enum
https://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html

**Pull request closed without merging**